### DISCUSSION TITLE
Consider adding support for NDJSON (aka JSONL) format

### DISCUSSION THREAD

**Discussion by ccoVeille** [Boeing/config-file-validator#133] [Ideas]
I would like you to consider [New Line delimited JSON](https://en.wikipedia.org/wiki/JSON_streaming) as a supported format.

The usual extensions for these file type are `.ndjson` or `.jsonnl`

I'm opening a discussion instead of opening a feature request/issue, because I would like to ask you first what you think about it

**Comment by ccoVeille** [comment]
For records, here is what a NDJSON look like

```json
{"value":"foo"}
{"value":"bar"}
```

Any "classic JSON file" is a one line NDJSON

**Comment by ccoVeille** [comment]
Here is a parser for ndjson with a lot of star

https://github.com/minio/simdjson-go


https://pkg.go.dev/github.com/minio/simdjson-go#ParseND

It could be useful

Even if parsing NDJSON is about trimming spaces, splitting by new lines then iterating on each line as a regular JSON

**Comment by kehoecj** [comment]
@ccoVeille Seems like a great addition - are you thinking updating the json validator to also support NDJSON or a separate validator?

**Reply by ccoVeille** [reply]
I would say a separate one, it would simplify the development, but you could have common way to validate a simple JSON or a multiple JSON line in a NDJSON file.

Your code rely on files extension. I don't think a `.json` file with multiple line of json (so a NDJSON file) should be named `.json`, but you may report the file structure is a ndjson. Instead of reporting it simply as invalid.

Here is what it currently reports
```
validator /tmp/multi.json
    × /tmp/multi.json
        error: Error at line 2 column 2: invalid character '{' after top-level value
```

**Reply by ccoVeille** [reply]
simdjson.ParseND seems limited to amd64, strange

https://github.com/minio/simdjson-go/blob/v0.4.5/parse_json_amd64.go
https://github.com/minio/simdjson-go/blob/v0.4.5/simdjson_other.go#L41

**Comment by ccoVeille** [comment]
@kehoecj could transform this discussion into an issue and assign it to me ? It would help me find them back on GitHub, thanks.
